containers/unit_tests/TestWithoutInitializing.hpp

```diff
@@ -437,21 +437,46 @@ TEST(TEST_CATEGORY, create_mirror_no_init_dynrankview) {
   listen_tool_events(Config::DisableAll(), Config::EnableKernels());
   Kokkos::DynRankView<int, TEST_EXECSPACE> device_view("device view", 10);
   Kokkos::DynRankView<int, Kokkos::HostSpace> host_view("host view", 10);
+  using MemorySpace = typename TEST_EXECSPACE::memory_space;
 
   auto success = validate_absence(
       [&]() {
         auto mirror_device =
             Kokkos::create_mirror(Kokkos::WithoutInitializing, device_view);
         ASSERT_EQ(device_view.size(), mirror_device.size());
+#ifndef KOKKOS_ENABLE_CUDA_UVM
+        static_assert(
+            std::is_same_v<typename decltype(mirror_device)::memory_space,
+                           Kokkos::HostSpace>);
+#else
+        static_assert(
+            std::is_same_v<typename decltype(mirror_device)::memory_space,
+                           MemorySpace>);
+#endif
```

Comment:
This is not the right condition to check. It happens to work but is not the right thing. You should check that if the device_view is host accessible, then its the same memory space, otherwise its hostspace.

Reply:
I would say the test is designed for this to be the right thing to check but sure I'm happy to change the condition.

Reply:
@crtrott Changed in [`9c62e10` (#5488)](https://github.com/kokkos/kokkos/pull/5488/commits/9c62e102b8015cd0edd5f234dc2f50f66bbffa77) and [`0ed8de1` (#5488)](https://github.com/kokkos/kokkos/pull/5488/commits/0ed8de1abbf546c60eb675e2d1270f934b9af97d).